src/caffe/proto/caffe.proto

```diff
@@ -445,6 +450,40 @@ message DataParameter {
   optional bool mirror = 6 [default = false];
 }
 
+// Message that stores parameters used by DataLayer
+message DatumDBParameter {
+  enum Mode {
+    NEW = 0;
+    WRITE = 1;
+    READ = 2;
+  }
+  // Specify the datumDB Backend.
+  optional string backend = 1 [default = "leveldb"];
+  // Specify the datumDB mode.
+  optional Mode mode = 2 [default = READ];
+  // Specify the datumDB source.
+  optional string source = 3;
+  // If true it will loop around the datumDB when needed
+  optional bool loop = 4 [default = true];
```

Comment:
This is sorely needed. When training, this should normally be true, and when testing, this should normally be false; I wonder if we should provide that default, or leave it up to the user's explicit specification? Anyway having the option is better than the status quo.